common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe.types
+
+import org.apache.spark.SparkException
+// scalastyle:off
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers
+
+import org.apache.spark.sql.catalyst.util.CollationFactory.getInstance
+import org.apache.spark.unsafe.types.UTF8String.{fromString => toUTF8}
+
+class CollationFactorySuite extends AnyFunSuite with Matchers { // scalastyle:ignore funsuite
```

Comment:
we should extend `SparkFunSuite` instead of `AnyFunSuite`

Reply:
Talked offline about this. We are staying with `AnyFunSuite` since this is in `common` and we `SparkFunSuite` here.